zookeeper/src/main/java/com/yahoo/ycsb/db/ZKClient.java

```diff
@@ -0,0 +1,188 @@
+ * <p>
+ * ZooKeeper client binding for YCSB.
+ * <p>
+ */
+
+package com.yahoo.ycsb.db;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+import java.util.Vector;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.zookeeper.CreateMode;
+import org.apache.zookeeper.KeeperException;
+import org.apache.zookeeper.WatchedEvent;
+import org.apache.zookeeper.Watcher;
+import org.apache.zookeeper.ZooDefs;
+import org.apache.zookeeper.ZooKeeper;
+import org.json.simple.JSONValue;
+
+import com.yahoo.ycsb.ByteIterator;
+import com.yahoo.ycsb.DB;
+import com.yahoo.ycsb.DBException;
+import com.yahoo.ycsb.Status;
+import com.yahoo.ycsb.StringByteIterator;
+
+/**
+ * YCSB binding for <a href="https://zookeeper.apache.org/">ZooKeeper</a>.
+ *
+ * See {@code zookeeper/README.md} for details.
+ */
+public class ZKClient extends DB {
+
+  private ZooKeeper zk;
+  private Watcher watcher;
+
+  public static final String CONNECT_STRING = "zookeeper.connectString";
+  public static final String DEFAULT_CONNECT_STRING = "127.0.0.1:2181";
+  public static final String SESSION_TIMEOUT_PROPERTY = "zookeeper.sessionTimeout";
+  public static final long DEFAULT_SESSION_TIMEOUT = TimeUnit.SECONDS.toMillis(30L);
+  public static final String WATCH_FLAG = "zookeeper.watchFlag";
+
+  public void init() throws DBException {
+    Properties props = getProperties();
+
+    String connectString = props.getProperty(CONNECT_STRING);
+    if (connectString == null || connectString.length() == 0) {
+      connectString = DEFAULT_CONNECT_STRING;
+    }
+
+    String watchFlagString = props.getProperty(WATCH_FLAG);
+    if (watchFlagString == null || "false".equals(watchFlagString)) {
+      watcher = null;
+    } else if ("true".equals(watchFlagString)) {
+      watcher = new SimpleWatcher();
+    } else {
+      throw new IllegalArgumentException("zookeeper.watchFlag only has two options:true or false");
+    }
+
+    long sessionTimeout;
+    String sessionTimeoutString = props.getProperty(SESSION_TIMEOUT_PROPERTY);
+    if (sessionTimeoutString != null) {
+      sessionTimeout = Integer.parseInt(sessionTimeoutString);
+    } else {
+      sessionTimeout = DEFAULT_SESSION_TIMEOUT;
+    }
+
+    try {
+      zk = new ZooKeeper(connectString, (int) sessionTimeout, null);
+    } catch (IOException e) {
+      throw new DBException("Closing connection failed.");
+    }
+  }
+
+  public void cleanup() throws DBException {
+    try {
+      zk.close();
+    } catch (InterruptedException e) {
+      throw new DBException("Closing connection failed.");
+    }
+  }
+
+  @Override
+  public Status read(String table, String key, Set<String> fields,
+                     Map<String, ByteIterator> result) {
+    String path = getPath(key);
+    try {
+      zk.getData(path, watcher, null);
+      return Status.OK;
+    } catch (KeeperException | InterruptedException e) {
+      e.printStackTrace();
```

Comment:
use a logging system and log a message w/this exception instead of calling `printStackTrace`. e.g. slf4j.